ext/openssl/ossl_ssl.c

```diff
@@ -1502,6 +1502,31 @@ ossl_ssl_s_alloc(VALUE klass)
     return TypedData_Wrap_Struct(klass, &ossl_ssl_type, NULL);
 }
 
+static VALUE
+peer_ip_address(VALUE self) {
+    VALUE io = rb_funcall(self, rb_intern("to_io"), 0);
+    VALUE remote_address = rb_funcall(io, rb_intern("remote_address"), 0);
+    
+    if (!RTEST(rb_funcall(remote_address, rb_intern("ip?"), 0))) {
+        return Qnil;
+    }
+
+    return rb_funcall(remote_address, rb_intern("ip_address"), 0);
+}
+
+static const char*
+peeraddr_ip_str(VALUE self) {
+    VALUE rb_mErrno = rb_const_get(rb_cObject, rb_intern("Errno"));
+    VALUE rb_eEINVAL = rb_const_get(rb_mErrno, rb_intern("EINVAL"));
+    VALUE rb_eENOTCONN = rb_const_get(rb_mErrno, rb_intern("ENOTCONN"));
+    VALUE rb_eECONNABORTED = rb_const_get(rb_mErrno, rb_intern("ECONNABORTED"));
+    VALUE address = rb_rescue2(peer_ip_address, self, NULL, (VALUE)0, rb_eEINVAL, rb_eENOTCONN, rb_eECONNABORTED, NULL);
```

Comment:
How can `Errno::ECONNABORTED` happen?

Since it's in error handling code anyway, would it make sense to just suppress any error rather than select error kinds, which can potentially differ between platforms?

Reply:
I'm not sure, I actually caught that error in CI. I agree, it makes more sense to rescue from any `SystemCallError`. I changed to rescue from it.